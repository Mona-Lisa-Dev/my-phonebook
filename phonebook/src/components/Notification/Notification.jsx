import PropTypes from 'prop-types';
import styles from './Notification.module.scss';

const Notification = ({ message, type }) => {
  const classNameType = type === 'error' ? styles.error : styles.notice;

  return <p className={classNameType}>{message}</p>;
};

Notification.defaultProps = {
  type: 'notice',
};

Notification.propTypes = {
  message: PropTypes.string,
  type: PropTypes.string,
};

export default Notification;
