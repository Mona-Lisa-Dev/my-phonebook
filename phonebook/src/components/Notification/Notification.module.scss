.Notification {
  margin: 20px 10px;
  text-align: center;
}

.error {
  @extend .Notification;
  color: red;
}

.notice {
  @extend .Notification;
  color: black;
}
