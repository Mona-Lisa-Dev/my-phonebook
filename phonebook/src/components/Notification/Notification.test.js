import { render } from '@testing-library/react';
import Notification from './Notification';

describe('Notification', () => {
  test('the first test', () => {
    const { container } = render(<Notification message="Oops" type="notice" />);

    expect(container).toMatchSnapshot();
  });
});
