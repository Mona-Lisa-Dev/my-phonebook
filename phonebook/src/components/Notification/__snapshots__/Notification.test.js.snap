// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification the first test 1`] = `
<div>
  <p
    class="notice"
  >
    Oops
  </p>
</div>
`;
